packages/frontend/core/src/components/blocksuite/block-suite-editor/specs/custom/attachment-block.ts

```diff
@@ -14,7 +19,15 @@ class CustomAttachmentBlockService extends AttachmentBlockService {
   }
 }
 
-export const CustomAttachmentBlockSpec: BlockSpec = {
+export const CustomAttachmentBlockSpec: ExtensionType[] = [
   ...AttachmentBlockSpec,
-  service: CustomAttachmentBlockService,
-};
+  {
+    setup: di => {
+      di.override(
+        BlockServiceIdentifier('affine:attachment'),
+        CustomAttachmentBlockService,
+        [StdIdentifier, BlockFlavourIdentifier('affine:attachment')]
```

Comment:
Can we improve and simplify this override API like `di.override('affine:attachment').to(CustomAttachmentBlockService)` in the feature?

Reply:
We will remove block services